packages/indexer/src/plugins/persistence.ts

```diff
@@ -0,0 +1,89 @@
+import type { Cursor } from "@apibara/protocol";
+import { type Database, type ISqlite, open } from "sqlite";
+import { defineIndexerPlugin } from "./config";
+
+type SqliteArgs = ISqlite.Config;
+
+export function sqlitePersistence<TFilter, TBlock, TRet>(args: SqliteArgs) {
```

Comment:
BTW now that we have a mock stream, we can test plugins end to end.

Add a tests where you setup a mock indexer with the `sqlitePersistence` plugin, run it over a sequence of messages and then assert that the stored cursor is the last `endCursor` in your stream.